### PR TITLE
✨ Backport Provider to release-0.7

### PR DESCRIPTION
**Pull request by Jeremy-Boyle**
**What this PR does / why we need it**:
This back ports the Provider feature into `v1alpha6`, numerous users would like the feature however `main` is not in a state to push a release.

Back port is based on PR: https://github.com/kubernetes-sigs/cluster-api-provider-openstack/pull/1501

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**TODOs**:

- [x] squashed commits
- [x] Push back port to Main as provider has been moved to `v1alpha6` PR: https://github.com/kubernetes-sigs/cluster-api-provider-openstack/pull/1530
- [ ] Create Issue / Feature to track this work between the two merges.
- if necessary:
  - [ ] includes documentation
  - [x] adds unit tests

/hold

### CODE REVIEW

**Comment by jichenjc** [comment]
usually we don't want to backport features to older release, mostly we backport bugs.. @mdbooth @seanschneeweiss any thoughts?

**Comment by ching-kuo** [comment]
@jichenjc There's a discussion on slack about backporting this feature to release 0.7
https://kubernetes.slack.com/archives/CFKJB65G9/p1682078907625039

**Comment by jichenjc** [comment]
Thanks for sharing , didn't notice that :(

**Comment by mdbooth** [comment]
> usually we don't want to backport features to older release, mostly we backport bugs.. @mdbooth @seanschneeweiss any thoughts?

I'd normally agree, but:
* we've got a few people waiting for this feature
* it's a relatively simple backport
* 0.8 might be a way out still

So we thought this was a pragmatic way forward.

**Comment by mdbooth** [comment]
/lgtm

I'll approve this after https://github.com/kubernetes-sigs/cluster-api-provider-openstack/pull/1530 lands in main.

**Review by dulek — APPROVED** [review]
Looks good! I pointed out something I noticed inline, but this is part of the original patch, so obviously no need to fix it here.

File: pkg/cloud/services/loadbalancer/loadbalancer.go [hunk 1]
```diff
@@ -351,6 +362,13 @@ func (s *Service) getOrCreateMonitor(openStackCluster *infrav1.OpenStackCluster,
 		Timeout:        5,
 	}
 	monitor, err = s.loadbalancerClient.CreateMonitor(monitorCreateOpts)
+	// Skip creating monitor if it is not supported by Octavia provider
+	if capoerrors.IsNotImplementedError(err) {
+		record.Warnf(openStackCluster, "SkippedCreateMonitor", "Health monitors are not supported with the current Octavia provider.")
+		record.Eventf(openStackCluster, "SkippedCreateMonitor", "Health Monitor is not created as it's not implemented with the current Octavia provider.")
```
[the remark this answers is not in the document]
Comment (dulek) [reply pkg/cloud/services/loadbalancer/loadbalancer.go:368]: Yup, it should be kept exactly like this. We can fix this on master and then backport the fix.

**Comment by Jeremy-Boyle** [comment]
https://github.com/kubernetes-sigs/cluster-api-provider-openstack/pull/1530 

is now in main

File: pkg/cloud/services/loadbalancer/loadbalancer.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (mdbooth) [reply pkg/cloud/services/loadbalancer/loadbalancer.go:368]: @dulek Do you want to submit that while it's fresh in your mind?

[the remark this answers is not in the document]
Comment (dulek) [reply pkg/cloud/services/loadbalancer/loadbalancer.go:368]: #1531

**Pull request merged**